### PR TITLE
update json schemas

### LINKED ISSUE #1486: Inconsistency between spec and schema
As per schema https://github.com/w3c/vc-data-model/blob/0e6d623d15fb6a3ffadf1249580b5ba55d7894e4/schema/verifiable-credential/verifiable-credential-schema.json#L210C6-L225, `credentialSchema` does not allow an array. It differs from other fields   such as `refreshService` or `termsOfUse` where both an array or object are allowed.

> ```
>     "credentialSchema": {
>       "type": "object",
>       "properties": {
>         "id": {
>           "type": "string"
>         },
>         "type": {
>           "$ref": "#/$defs/type"
>         }
>       },
>       "required": [
>         "id",
>         "type"
>       ],
>       "additionalProperties": true
>     },
> ```

However as per the spec in [example 22](https://www.w3.org/TR/vc-data-model-2.0/#example-usage-of-the-credentialschema-property-to-perform-json-schema-validation) and https://www.w3.org/TR/vc-data-model-2.0/#data-schemas it states

> The value of the credentialSchema [property](https://www.w3.org/TR/vc-data-model-2.0/#dfn-property) MUST be one or more data schemas...

Please clarify which is correct, the `schema.json` or the written document.

### PR DESCRIPTION
**Pull request by decentralgabe**
Fix #1486 

- support arrays for credential status, credential schema, evidence
- fix bug in representing type (can be a single value or an array)
- added regex for xml date time

### CODE REVIEW

File: schema/verifiable-credential/verifiable-credential-schema.json [hunk 1]
```diff
@@ -172,7 +212,12 @@
       ]
     },
     "validFrom": {
-      "type": "string"
+      "type": "string",
+      "pattern": ".*(Z|(\\+|-)[0-9][0-9]:[0-9][0-9])"
+    },
+    "validUntil": {
+      "type": "string",
+      "pattern": ".*(Z|(\\+|-)[0-9][0-9]:[0-9][0-9])"
```
Comment (iherman) [inline schema/verifiable-credential/verifiable-credential-schema.json:220]: Are we sure about this regex? Valid from/until are supposed to be dates, and these only seem to restrict the time part. Why not using a regex for the full xsd datatype?

File: schema/verifiable-presentation/verifiable-presentation-schema.json [hunk 2]
```diff
@@ -56,7 +53,8 @@
           ]
         },
         "created": {
-          "type": "string"
+          "type": "string",
+          "pattern": ".*(Z|(\\+|-)[0-9][0-9]:[0-9][0-9])"
```
Comment (iherman) [inline schema/verifiable-presentation/verifiable-presentation-schema.json:57]: Just marking with the same comment as above

File: schema/verifiable-credential/verifiable-credential-schema.json (hunk 1 above)
Comment (msporny) [reply schema/verifiable-credential/verifiable-credential-schema.json:220]: IIUC, these regexes are incorrect. The correct regex can be found here:

https://www.w3.org/TR/vc-data-model-2.0/#example-regular-expression-to-detect-a-valid-xml-schema-1-1-part-2-datetimestamp

File: schema/verifiable-presentation/verifiable-presentation-schema.json (hunk 2 above)
Comment (msporny) [reply schema/verifiable-presentation/verifiable-presentation-schema.json:57]: Same issue, use the regex here instead:

https://www.w3.org/TR/vc-data-model-2.0/#example-regular-expression-to-detect-a-valid-xml-schema-1-1-part-2-datetimestamp

**Review by msporny — CHANGES_REQUESTED** [review]
Regexes for XMLDatetimeStamp are incorrect. Otherwise, LGTM.

**Review by dlongley — APPROVED** [review]
Approving assuming the regexes will be updated as suggested.

**Comment by decentralgabe** [comment]
thanks @iherman @msporny updated the regex!

**Review by iherman — APPROVED** [review]

**Review by msporny — APPROVED** [review]

**Review by TallTed — APPROVED** [review]

**Comment by decentralgabe** [comment]
changes addressed; opened over a week; editorial approval; merging!

**Pull request merged**